hathor/types.py

```diff
@@ -21,3 +24,41 @@
 Timestamp = int         # NewType('Timestamp', int)
 TokenUid = VertexId     # NewType('TokenUid', VertexId)
 Amount = int            # NewType('Amount', int)
+
+
+class BlockInfo(NamedTuple):
+    hash_hex: str
+    height: int
+    weight: float
+
+    @staticmethod
+    def from_raw(block_info_raw: tuple[str, int, float]) -> 'BlockInfo':
+        """ Instantiate BlockInfo from a literal tuple.
+        """
+        if not (isinstance(block_info_raw, list) and len(block_info_raw) == 3):
+            raise ValueError(f"block_info_raw must be a tuple with length 3. We got {block_info_raw}.")
+
+        hash_hex, height, weight = block_info_raw
+
+        if not isinstance(hash_hex, str):
+            raise ValueError(f"hash_hex must be a string. We got {hash_hex}.")
+        hash_pattern = r'[a-fA-F\d]{64}'
+        if not re.match(hash_pattern, hash_hex):
+            raise ValueError(f"hash_hex must be valid. We got {hash_hex}.")
+        if not isinstance(height, int):
+            raise ValueError(f"height must be an integer. We got {height}.")
+        if height < 0:
+            raise ValueError(f"height must greater than or equal to 0. We got {height}.")
+        if not isinstance(weight, (float, int)):
+            raise ValueError(f"weight must be a float. We got {weight}.")
+        if not weight > 0:
+            raise ValueError(f"weight must be greater than 0. We got {weight}.")
+
+        return BlockInfo(hash_hex, height, weight)
+
+    def __eq__(self, other):
```

Comment:
Why do you need it? As it is a named tuple, I feel this is exactly what it does by default.